internal/trivy/trivy.go

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+*
+* Copyright 2023 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package trivy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// see https://github.com/aquasecurity/trivy/blob/main/pkg/flag/remote_flags.go#L11
+const (
+	TokenHeader = "Trivy-Token"
+	KeppelToken = "Keppel-Token"
+)
+
+type Config struct {
+	URL   url.URL
+	Token string
+}
+
+type securityRespone struct {
+	Results []struct {
+		Vulnerabilities []struct {
+			Severity string `json:"Severity"`
+		} `json:"Vulnerabilities"`
+	} `json:"Results"`
+}
+
+func (tc *Config) ScanManifest(manifestRefString, keppelToken string) (trivyReport securityRespone, returnedError error) {
+	//we don't allow Trivy to take more than 10 minutes on a single image (which is already an
+	//insanely generous timeout)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	requestURL := tc.URL
+	requestURL.Path = "/trivy"
+	requestURL.RawQuery = "image=" + manifestRefString
```

Comment:
I don't have an immediate reason to suggest that `manifestRefString` could contain special chars that break the query syntax, but it's probably still a good idea to be defensive here.

```suggestion
	requestURL.RawQuery = url.Values{"image": {manifestRefString}}.Encode()
```